frontend/src/metabase/nav/components/SearchBar.tsx

```diff
@@ -57,6 +71,8 @@ function SearchBarView({ location, onSearchActive, onSearchInactive }: Props) {
   const searchInput = useRef<HTMLInputElement>(null);
   const dispatch = useDispatch();
 
+  const hasSearchText = searchText.trim().length > 0;
```

Comment:
`typeof searchText === "string"` is removed. I'm not sure if this will crash. If not that means that type check doesn't do anything in the past and `searcText` is always a string.

Reply:
Yeah, I don't know what that type check was doing - `location.query.q` (which contains the search text in the query params) is typed as `string | undefined`, and we return an empty string if `q` is `undefined`. From my understanding `searchText` cannot be anything else besides a string

Reply:
Sounds good to me then. But please keep in mind that sometimes we want to be able to set the value to `null` e.g. when removing a dashboard description.

It seems in this case we only call `setSearchText` with:
1. `e.target.value`
2. `""`
3. `getSearchTextFromLocation(location)`

which should always be a string, so it's fine. But if, for instance, the `searchText` relies on a prop, it might be able to turn back to `null` or `undefined` in some cases as I've mentioned above.